openquake/hazardlib/site.py

```diff
@@ -56,37 +53,36 @@ class Site(object):
 
         :class:`Sites <Site>` are pickleable
     """
-    _slots_ = 'location vs30 vs30measured z1pt0 z2pt5 backarc'.split()
-
-    def __init__(self, location, vs30, vs30measured, z1pt0, z2pt5,
-                 backarc=False):
-        if not vs30 > 0:
+    def __init__(self, location, vs30=numpy.nan, vs30measured=False,
+                 z1pt0=numpy.nan, z2pt5=numpy.nan, **extras):
```

Comment:
To some extent, `z1pt0`, `z2pt5` and `vs30measured` are not so different from other optional attributes, in the sense that only a couple of GMPEs use them, so they would not be critical for every calculation but only those using those particular GMPEs, and it can often be a nuisance to the users to have to set values for these when they are not being used. But, I guess these have been a legacy in OQ for it's entire life, so changing that would probably require some effort.

Reply:
Indeed, @g-weatherill , it is just a matter of legacy. I do not want to break all the code instantiating the `Site` object. For `backarc` it is not an issue, since it is a relatively "new" parameter and it has a default, while the others have been mandatory parameters since the beginning of time.